### PR TITLE
V1 to V2 deployment scripts

### LINKED ISSUE #115: E2E Migration test

### PR DESCRIPTION
**Pull request by pahor167**
### Description

Update of deployment scripts for V2. Introduction of new hardhat task for update from V1 to V2

### Other changes

No other changes

### Tested

Mostly manually since there are issues with devchain and alfajores doesn't have latest core contract updates.

### Related issues

- Fixes #115

### CODE REVIEW

File: deploy/00_multisig.ts [hunk 1]
```diff
@@ -36,11 +37,53 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
...
+        if (parsedTx.functionFragment.name === "upgradeTo") {
+          console.log(
+            chalk.red(
+              "Transaction was reverted. Most probably it was because caller is not an owner. Please make sure to update the proxy implementation manually."
+            )
+          );
+          return;
+        }
+      }
+      console.log(e);
```
Comment (montera82) [inline deploy/00_multisig.ts:85]: do we need this?

File: deploy/00_multisig.ts [hunk 2]
```diff
@@ -36,11 +37,53 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
...
+    if (e?.data?.stack?.indexOf("VM Exception while processing transaction: revert")) {
+      console.log(
+        chalk.red(
+          "Transaction was reverted. Most probably it was because caller is not an owner. Please make sure to update the proxy implementation manually."
+        )
+      );
+    } else {
+      const multisigContract = await hre.ethers.getContractOrNull("MultiSig");
+
+      if (multisigContract != null && e.transaction != null) {
```
Comment (montera82) [inline deploy/00_multisig.ts:71]: instead of `e` can we use the name err ?

File: lib/deploy-utils.ts [hunk 3]
```diff
@@ -18,24 +19,39 @@ export async function catchNotOwnerForProxy(
...
+          "Transaction was reverted since caller is not an owner. Please make sure to update the proxy implementation manually."
+        )
       );
       return;
     } else if (e.error?.data) {
       const data = e.error.data;
       const encodedSenderMustBeMultisigWallet = web3.eth.abi.encodeFunctionSignature(
         "SenderMustBeMultisigWallet(address)"
       );
+      console.log("encodedSenderMustBeMultisigWallet", encodedSenderMustBeMultisigWallet);
```
Comment (montera82) [inline lib/deploy-utils.ts:36]: do we need this console?

**Review by soloseng — APPROVED** [review]
🚀

**Pull request merged**